src/MCPClient/lib/clientScripts/create_sip_from_transfer_objects.py

```diff
@@ -85,6 +85,19 @@ def call(jobs):
                         lookup_path, UUID=sip_uuid, sip_type=sip_type, diruuids=diruuids, printfn=job.pyprint)
                     sip = SIP.objects.get(uuid=sip_uuid)
 
+                # Set activeAgent using the value in Transfer. This ensures
+                # that events generated in Ingest can fall to this value in
+                # scenarios where the processing config does not require user
+                # interfactions, e.g. in the "automated" processing config.
+                try:
+                    agent = Agent.objects.get(id=UnitVariable.objects.get(
+                        unittype="Transfer", unituuid=transferUUID,
+                        variable="activeAgent").variablevalue)
+                except (Agent.DoesNotExist, UnitVariable.DoesNotExist):
```

Comment:
Nitpicky but I would break this out
```suggestion
                try:
                    unit_variable = UnitVariable.objects.get(
                        unittype="Transfer", unituuid=transferUUID, variable="activeAgent")
                except UnitVariable.DoesNotExist:
                    unit_variable = None
                
                if unit_variable:
                    try:
                        agent = Agent.objects.get(id=unit_variable.variablevalue)
                    except Agent.DoesNotExist:
```
(or alternately, get rid of the if and lookup the agent in the else clause of the try/except).
It's a bit clunkier (I guess, depends on your perspective) but I don't like having multiple lookups per statement.